Added custom engine version support in the integration tests

miazamrai opened the pull request:
Integration tests can honor an environment variable like  CUSTOM_HEADERS= '{"key1":"val1", "key2":"val2"}'. The custom headers key value pair should be valid JSON string otherwise, I runtime error will be thrown. The JSON string will be parsed into a dictionary and will passed to the create_engine function as kwargs...


export CUSTOM_HEADERS='{"x-rai-parameter-compute-version":"6df2736eccdc6a4844c2f8e8d78bcd7782046e1d"}'

bachdavi reviewed (commented):
Cool :)

Could you add an example of a valid test argument to the comments? 
`parse_test_args` will parse the args of the `julia` executable not from a julia REPL, right?

I have added the example but I am not sure whether parse_test_args will work with REPL or not? I am not that familiar with Julia yet :)

miazamrai commented:
> I have added the example but I am not sure whether parse_test_args will work with REPL or not? I am not that familiar with Julia yet :)
> 
> The github action is calling it using julia binary with --project



> Cool :)
> 
> Could you add an example of a valid test argument to the comments? `parse_test_args` will parse the args of the `julia` executable not from a julia REPL, right?

I have added the example but I am not sure whether parse_test_args will work with REPL or not? I am not that familiar with Julia yet :)

The github action is calling it using julia binary with --project

NRHelmi reviewed (approved):
Thanks @miazamrai

bachdavi commented:
> > I have added the example but I am not sure whether parse_test_args will work with REPL or not? I am not that familiar with Julia yet :)
> > The github action is calling it using julia binary with --project
> 
> > Cool :)
> > Could you add an example of a valid test argument to the comments? `parse_test_args` will parse the args of the `julia` executable not from a julia REPL, right?
> 
> I have added the example but I am not sure whether parse_test_args will work with REPL or not? I am not that familiar with Julia yet :)
> 
> The github action is calling it using julia binary with --project
> 
> It also works with REPL Like julia> include("test/runtests.jl") -> goes to the create engine without version

Awesome 🎉 
Could you add the two to the source code as well :)
I just checked `ArgParse.jl` and we use the `Base.ARGS` constant, a vector of strings containing the command line arguments. It seems this cannot be changed in a running REPL...
But I guess that is fine!

bachdavi commented:
Also we see this in the test logs: 
```
┌ Warning: `@add_arg_table` is deprecated, use `@add_arg_table!` instead
└ @ ArgParse ~/work/rai-sdk-julia/rai-sdk-julia/test/runtests.jl:34
```
We can change it to the one with the bang.

miazamrai commented:
> > I have added the example but I am not sure whether parse_test_args will work with REPL or not? I am not that familiar with Julia yet :)
> > The github action is calling it using julia binary with --project
> 
> > Cool :)
> > Could you add an example of a valid test argument to the comments? `parse_test_args` will parse the args of the `julia` executable not from a julia REPL, right?
> 
> I have added the example but I am not sure whether parse_test_args will work with REPL or not? I am not that familiar with Julia yet :)
> 
> The github action is calling it using julia binary with --project

Tested it with REPL, it works!

Like 

julia> include(test/runtests.jl)

or

module test
       ARGS={"}
       include("test.jl")
       end

> Also we see this in the test logs:
> 
> ```
> ┌ Warning: `@add_arg_table` is deprecated, use `@add_arg_table!` instead
> └ @ ArgParse ~/work/rai-sdk-julia/rai-sdk-julia/test/runtests.jl:34
> ```
> 
> We can change it to the one with the bang.

@bachdavi I have talked to @NRHelmi and we are agreed on sending the custom headers as an env variable. So no command line arguments will be used. It is to use the same approach for all the SDKs and env variable seems like the easiest and the most consistent approach to follow.
